wordpress-mobile/WordPress-iOS
# Feature/4105 sharing management

Refs #4105 

This is a partial implementation of Publicize sharing management presenting the overall publicize flow. This PR is submitted in the interest of keeping code reviews to relatively small sets of code. Sharing remains behind a feature flag.
Remaining work includes:
- implementing some missing graphics or changing placeholders.
- implement some designs around multiple account connections once the designs are complete.
- improvements to error / success prompts and "busy" status 
- all of sharing button management.

To test:

Choose a wpcom blog and visit its details screen. 
Select the Sharing option. 
For each publicize service perform the following:
- Connect to the service.
- Confirm the service is properly connected by checking the site's publicize settings in a browser. 
- Publish a post and confirm it is shared to the target service. (Optional, we assume publicize works correctly.)
- Toggle the share globally setting and confirm the change is reflected on the web.
- Disconnect from the service. 
- Confirm the service is disconnected by checking the site's publicize settings in a browser. 

Needs review: @jleandroperez would you do the honors?
